app/app.go

```diff
@@ -184,6 +184,8 @@ func (a *App) RunAndWait(mainLoop MainLoopFunc) {
 	errs := make(chan error)
 
 	go func() {
+		defer Recover()
```

Comment:
Instead of a log fatal here, I would prefer returning an error. So that it will trigger a gracefulshutdown. Please read the app/v2 [proposal](https://github.com/arquivei/foundationkit/blob/app-v2/app/v2/app.go#L206)

```go
	defer func() {
		if r := recover(); r != nil {
			errs <- errors.NewFromRecover(r)
		}
	}()
```